coverage/data.py

```diff
@@ -82,12 +82,17 @@ def combinable_files(data_file: str, data_paths: Iterable[str] | None = None) ->
             pattern = glob.escape(os.path.join(os.path.abspath(p), local)) +".*"
             files_to_combine.extend(glob.glob(pattern))
         else:
-            raise NoDataError(f"Couldn't combine from non-existent path '{p}'")
+            raise DataFileOrDirectoryNotFoundError.new_for_data_file_or_directory(
```

Comment:
I think I understand why you are using classmethods instead of the constructor, but let's just call it `new`.  The method name pretty much duplicates the class name, and there is only one classmethod.